src/adapter/src/session/vars.rs

```diff
@@ -1174,3 +1183,73 @@ impl Value for TimeZone {
         self.as_str().into()
     }
 }
+
+/// List of valid isolation levels.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum IsolationLevel {
+    ReadUncommitted,
+    ReadCommitted,
+    RepeatableRead,
+    Serializable,
+    StrictSerializable,
+}
+
+impl IsolationLevel {
+    pub(super) fn as_str(&self) -> &'static str {
+        match self {
+            Self::ReadUncommitted => "READ_UNCOMMITTED",
+            Self::ReadCommitted => "READ_COMMITTED",
+            Self::RepeatableRead => "REPEATABLE_READ",
+            Self::Serializable => "SERIALIZABLE",
+            Self::StrictSerializable => "STRICT_SERIALIZABLE",
+        }
+    }
+
+    fn valid_values() -> Vec<&'static str> {
+        vec![
+            Self::ReadUncommitted.as_str(),
+            Self::ReadCommitted.as_str(),
+            Self::RepeatableRead.as_str(),
+            Self::Serializable.as_str(),
+            Self::StrictSerializable.as_str(),
+        ]
+    }
+}
+
+impl Value for IsolationLevel {
+    const TYPE_NAME: &'static str = "string";
+
+    fn parse(s: &str) -> Result<Self::Owned, ()> {
```

Comment:
This looks like it allows users to ask for levels below serializable and that those will be accepted and printed if asked. I think we want to accept them, but always bump up to serializable? There's things at tension here: at least one ORM asks for some low level, and we should either say nope (breaking the ORM) or say yes but then actually change it. I think accepting and changing matches our current behavior, so let's try that?

Reply:
What are the trade-offs here? 
1. If a person sets a thing equal to `foo`, do they expect it to be equal to `foo`? Like, might someone set it, believe it is set, and then either error, or retry, or something until it is actually set to their value? 
2. If a person uses a low setting, but could benefit from knowing they would get stronger guarantees, how should we communicate that? By changing their level to serializable?

Seems like this is partly about norms of users, and whether that constrains us?

Reply:
Done. I'll just make sure to make this clear in the documentation.

Reply:
FWIW cockroach complains if asked for less than serializable. We chose to silently bump up for ORM/bi tool support. Cockroach probably doesn't need to do that because there's custom drivers for the tools they support. I think I'd prefer to error, but I'd prefer even more to just have some bi tools work out of the box, and I think the tradeoffs you list are acceptable. I think about this like users are asking us not to break stuff in known ways rather than "please go fast and break stuff if you want".

Reply:
I think you highlighted the trade-off well:
1. With the approach I just changed it to, it may be confusing if a user sets the isolation level to Read Committed, but it keeps showing as Serializable.
2. With the original approach, if a user sees that the isolation level is Read Committed, they may expect some performance improvements over Serializable.

Reply:
Ideally, we just print a warning along the lines of
> All isolation levels below Serializable will automatically get upgraded to Serializable.